docs/developer-guide/working-with-plugins.md

```diff
@@ -18,6 +18,25 @@ module.exports = {
 };
 ```
 
+### Preprocessors in Plugins
+
+You can also create plugins that would tell ESLint how to process files other then JavaScript. In order to create a
+preprocessor, object that is exported from your module has to conform to the following interface:
+
+```js
+preprocessors: {
```

Comment:
Up do you, I can change the name to be `processors`, I was calling them `preprocessors` because that's a pretty standard term for a script that does something to the code before sending it to CLI (in this case, remove everything that's not JavaScript)